File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.js
function solution(str_list) {
    const li = str_list.indexOf('l');
    const ri = str_list.indexOf('r');
    
    if (li === -1 && ri === -1) return [];
    
    if (li !== -1 && (ri === -1 || li < ri)) {
        return str_list.slice(0, li);
    } else if (ri !== -1 && (li === -1 || ri < li)) {
        return str_list.slice(ri + 1);
    }
    
    return [];
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.java
import java.util.*;

class Solution {
    public String[] solution(String[] str_list) {
        List<String> list = Arrays.asList(str_list);
        
        int l_idx = list.indexOf("l");
        int r_idx = list.indexOf("r");
        
        if (l_idx < 0 && r_idx < 0) return new String[]{};
        
        if (r_idx < 0 || (l_idx != -1 && l_idx < r_idx)) {
            return list.subList(0, l_idx).toArray(new String[0]);
        }
        
        return list.subList(r_idx+1, list.size()).toArray(new String[0]);   
    }
}